frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -850,11 +857,18 @@ export function stream(url: string, cb: StreamResultsCb, args: StreamArgs) {
   function cancel() {
     if (connection) connection.close();
     isCancelled = true;
+    OPEN_SOCKET_CONNECTIONS.delete(url);
   }
 
   function connect() {
     if (connectCb) connectCb();
-    connection = connectStream(url, cb, onFail, isJson, additionalProtocols);
+    if (OPEN_SOCKET_CONNECTIONS.has(url)) {
+      const { socket, close } = OPEN_SOCKET_CONNECTIONS.get(url)!;
```

Comment:
I think this may not be enough to verify it's the same connect stream.
If I call the same URL but with a different callback, onFail, isJson, or additionalProtocols, then this code will not have the expected behavior.

Making URL+isJson+additionalProtocols the actual key for the map should be easy, but checking for the same cb+onFail may be tricky and it's not very efficient to have different connections just for different callbacks, so maybe a solution is to have a generic callback & onFail that will be in charge of calling the user set callback/onFail (which we keep track of). We must not forget to update this list of callbacks/onFails if the user has cancelled their connection.